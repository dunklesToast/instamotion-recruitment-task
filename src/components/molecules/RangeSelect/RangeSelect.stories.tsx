import React, { useState } from 'react';
import type { Meta, Story } from '@storybook/react';
import type { RangeChangeEvent, RangeSelectProps } from './RangeSelect';
import { RangeSelect } from './RangeSelect';

export default {
  title: 'Molecules/RangeSelect',
  component: RangeSelect,
} as Meta;

const RangeSelectComponent: Story<RangeSelectProps> = () => {
  const [start, setStart] = useState('');
  const [end, setEnd] = useState('');

  const update = (values: RangeChangeEvent) => {
    setStart(values.start);
    setEnd(values.end);
  };

  return <RangeSelect name='Storybook' start={start} end={end} onChange={update} />;
};

export const Default = RangeSelectComponent.bind({});
