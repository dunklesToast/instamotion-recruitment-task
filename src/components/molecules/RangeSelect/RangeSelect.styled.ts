import styled from 'styled-components';

export const Input = styled.input`
  border: 1px black solid;
  border-radius: 4px;
  font-family: 'PT Sans', sans-serif;
  margin-top: 4px;
  padding: 4px;
  width: 80px;
`;

export const Spacer = styled.span`
  margin: 0 8px;
`;
