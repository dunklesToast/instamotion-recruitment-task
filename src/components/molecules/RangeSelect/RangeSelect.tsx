import React from 'react';
import { Text } from '@atoms/Text/Text';
import * as S from './RangeSelect.styled';

export type RangeChangeEvent = {
  start: string;
  end: string;
};

export interface RangeSelectProps {
  /** Beginning Value */
  start: string;
  /** End Value */
  end: string;
  /** Human Readable Range Select Name */
  name: string;
  /** onChange */
  onChange: (e: RangeChangeEvent) => void;
}

/**
 * Description of RangeSelect.
 */
export function RangeSelect({ start, end, onChange, name }: RangeSelectProps): JSX.Element {
  return (
    <div data-testid='range-select'>
      <Text>{name}</Text>
      <S.Input
        type='text'
        placeholder='von'
        onChange={(e) =>
          onChange({
            start: e.target.value,
            end,
          })
        }
        value={start}
      />
      <S.Spacer>-</S.Spacer>
      <S.Input
        type='text'
        placeholder='bis'
        onChange={(e) =>
          onChange({
            start,
            end: e.target.value,
          })
        }
        value={end}
      />
    </div>
  );
}
