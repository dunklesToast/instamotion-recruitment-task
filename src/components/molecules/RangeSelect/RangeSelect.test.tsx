import React, { useState } from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import type { RangeChangeEvent } from './RangeSelect';
import { RangeSelect } from './RangeSelect';

const RangeSelectRenderer = ({ mock }) => {
  const [start, setStart] = useState('');
  const [end, setEnd] = useState('');

  const onChange = (e: RangeChangeEvent) => {
    setStart(e.start);
    setEnd(e.end);
    mock(e);
  };

  return <RangeSelect onChange={onChange} start={start} end={end} name='Test' />;
};

test('RangeSelect', async () => {
  const mock = jest.fn();
  render(<RangeSelectRenderer mock={mock} />);

  const element = () => screen.getByTestId('range-select');
  const inputFrom = () => screen.getByPlaceholderText('von');
  const inputTo = () => screen.getByPlaceholderText('bis');

  expect(element()).toBeInTheDocument();
  expect(element()).toMatchSnapshot();

  await userEvent.type(inputFrom(), '2');

  expect(mock).toBeCalledWith({
    start: '2',
    end: '',
  });

  mock.mockReset();

  await userEvent.type(inputTo(), '5');

  expect(mock).toBeCalledWith({
    start: '2',
    end: '5',
  });
});
