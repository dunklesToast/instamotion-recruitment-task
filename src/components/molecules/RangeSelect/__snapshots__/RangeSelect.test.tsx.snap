// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSelect 1`] = `
.c0 {
  font-family: 'PT Sans',sans-serif;
  font-style: regular;
  font-weight: 400;
}

.c1 {
  border: 1px black solid;
  border-radius: 4px;
  font-family: 'PT Sans',sans-serif;
  margin-top: 4px;
  padding: 4px;
  width: 80px;
}

.c2 {
  margin: 0 8px;
}

<div
  data-testid="range-select"
>
  <p
    class="c0"
  >
    Test
  </p>
  <input
    class="c1"
    placeholder="von"
    type="text"
    value=""
  />
  <span
    class="c2"
  >
    -
  </span>
  <input
    class="c1"
    placeholder="bis"
    type="text"
    value=""
  />
</div>
`;
